src/System.Net.Security/src/System/Net/Security/SslState.cs

```diff
@@ -1449,6 +1420,33 @@ private void FinishHandshake(Exception e, AsyncProtocolRequest asyncRequest)
             }
         }
 
+        private void ProcessPendingWrite()
+        {
+            object obj = _queuedWriteStateRequest;
+            _queuedWriteStateRequest = null;
+
+            switch (obj)
+            {
+                case null:
+                    // We finished before Write has grabbed the lock.
+                    return;
+                case LazyAsyncResult lazy:
+                    // Sync write is waiting on other thread.
+                    lazy.InvokeCallback();
+                    return;
+                case TaskCompletionSource<int> completionSource:
+                    //Async write is pending, should it be started on another thread?
+                    //Or is it fine to do it here
```

Comment:
What does this comment mean?

Reply:
Basically below they do a task.new () wonder if I should do similar?

Reply:
Oh, you mean do you need to queue the completion of it?  That depends: might there be continuations off of this that we don't want to run as part of this call?  If we do want to offload it, there are two options: if you only want to sometimes offload it, you can just queue the call to SetResult, as is done below.  But if you always want to make sure that continuations run asynchronously from the SetResult call, you can pass TaskCreationOptions.RunContinuationsAsynchronously to the TaskCompletionSource's ctor.

Reply:
I will go with the second one. We don't want the result from the action calling SetResult to be delayed by a write that is pending.